File: samples/tags/build 1.1.0.33/C#/MsaWithDa/MsaWithDa.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace MsaWithDa
{
	public partial class MsaWithDa : Form
	{
		const String mMsCharacterName = "Genie";
		const String mMsCharacterFile = "C:\\Windows\\MsAgent\\Chars\\Genie.acs";
		const String mDaCharacterName = "Merlin";
		const String mDaCharacterFile = "C:\\Windows\\MsAgent\\Chars\\Merlin.acs";

		AgentServerObjects.IAgentEx mMsServer;
		AgentServerObjects.IAgentCharacterEx mMsServerChar;
		int mMsServerCharId = 0;
		AgentObjects.IAgentCtlCharacterEx mMsControlChar;

		DoubleAgent.Server.Server mDaServer;
		DoubleAgent.Server.Character mDaServerChar;
		int mDaServerCharId = 0;
		DoubleAgent.Control.Control mDaControl;
		DoubleAgent.Control.Character mDaControlChar;

		public MsaWithDa()
		{
			InitializeComponent();

			mDaControl = new DoubleAgent.Control.Control ();
			mDaControl.Show += new DoubleAgent.Control.ShowEventHandler (mDaControl_Show);
			mDaControl.Hide += new DoubleAgent.Control.HideEventHandler (mDaControl_Hide);

			SetMsControlButtons();
			SetMsServerButtons();
			SetDaControlButtons();
			SetDaServerButtons();
		}

		private void SetMsControlButtons()
		{
			if (mMsControlChar == null)
			{
				mLoadMsControlChar.Enabled = true;
				mUnloadMsControlChar.Enabled = false;
				mShowMsControlChar.Enabled = false;
				mHideMsControlChar.Enabled = false;
			}
			else
			{
				mLoadMsControlChar.Enabled = false;
				mUnloadMsControlChar.Enabled = true;
				mShowMsControlChar.Enabled = !mMsControlChar.Visible;
				mHideMsControlChar.Enabled = mMsControlChar.Visible;
			}
		}

		private void SetMsServerButtons()
		{
			if (mMsServerChar == null)
			{
				mLoadMsServerChar.Enabled = true;
				mUnloadMsServerChar.Enabled = false;
				mShowMsServerChar.Enabled = false;
				mHideMsServerChar.Enabled = false;
			}
			else
			{
				mLoadMsServerChar.Enabled = false;
				mUnloadMsServerChar.Enabled = true;
				mShowMsServerChar.Enabled = true;
				mHideMsServerChar.Enabled = true;
			}
		}

		private void SetDaControlButtons()
		{
			if (mDaControlChar == null)
			{
				mLoadDaControlChar.Enabled = true;
				mUnloadDaControlChar.Enabled = false;
				mShowDaControlChar.Enabled = false;
				mHideDaControlChar.Enabled = false;
			}
			else
			{
				mLoadDaControlChar.Enabled = false;
				mUnloadDaControlChar.Enabled = true;
				mShowDaControlChar.Enabled = !mDaControlChar.Visible;
				mHideDaControlChar.Enabled = mDaControlChar.Visible;
			}
		}

		private void SetDaServerButtons()
		{
			if (mDaServerChar == null)
			{
				mLoadDaServerChar.Enabled = true;
				mUnloadDaServerChar.Enabled = false;
				mShowDaServerChar.Enabled = false;
				mHideDaServerChar.Enabled = false;
			}
			else
			{
				mLoadDaServerChar.Enabled = false;
				mUnloadDaServerChar.Enabled = true;
				mShowDaServerChar.Enabled = true;
				mHideDaServerChar.Enabled = true;
			}
		}

		private void MsControl_ShowEvent(object sender, AxAgentObjects._AgentEvents_ShowEvent e)
		{
			SetMsControlButtons();
		}

		private void MsControl_HideEvent(object sender, AxAgentObjects._AgentEvents_HideEvent e)
		{
			SetMsControlButtons();
		}

        private void mDaControl_Show(string CharacterID, DoubleAgent.Control.VisibilityCauseType Cause)
        {
            SetDaControlButtons();
        }

		private void mDaControl_Hide (string CharacterID, DoubleAgent.Control.VisibilityCauseType Cause)
        {
			SetDaControlButtons();
		}

		private void LoadMsServerChar_Click(object sender, EventArgs e)
		{
			int lReqId;
			object lCharacter;

			if (mMsServer == null)
			{
				mMsServer = (AgentServerObjects.IAgentEx)new AgentServerObjects.AgentServer();
			}

			if (mMsServerChar == null)
			{
				mMsServer.Load(mMsCharacterFile, out mMsServerCharId, out lReqId);
				mMsServer.GetCharacter(mMsServerCharId, out lCharacter);
				mMsServerChar = (AgentServerObjects.IAgentCharacterEx)lCharacter;
				if (mMsServerChar != null)
				{
					mMsServerChar.MoveTo(700, 100, 0, out lReqId);
				}
			}
			SetMsServerButtons();
		}

		private void UnloadMsServerChar_Click(object sender, EventArgs e)
		{
			if ((mMsServer != null) && (mMsServerChar != null))
			{
				mMsServerChar = null;
				mMsServer.Unload(mMsServerCharId);
				mMsServerCharId = 0;
			}
			SetMsServerButtons();
		}

		private void ShowMsServerChar_Click(object sender, EventArgs e)
		{
			int lReqId;

			if (mMsServerChar != null)
			{
				mMsServerChar.Show(0, out lReqId);
				mMsServerChar.Speak(mMsCharacterName + ", Microsoft Agent", "", out lReqId);
			}
			SetMsServerButtons();
		}

		private void HideMsServerChar_Click(object sender, EventArgs e)
		{
			int lReqId;

			if (mMsServerChar != null)
			{
				mMsServerChar.Hide(0, out lReqId);
			}
			SetMsServerButtons();
		}

		private void ShowMsServerOptions_Click(object sender, EventArgs e)
		{
			AgentServerObjects.IAgentPropertySheet lServerProps;

			if (mMsServer == null)
			{
				mMsServer = (AgentServerObjects.IAgentEx)new AgentServerObjects.AgentServer();
			}
			lServerProps = (AgentServerObjects.IAgentPropertySheet)mMsServer;
			lServerProps.SetVisible(1);
		}

		private void ShowMsServerChars_Click(object sender, EventArgs e)
		{
			if (mMsServer == null)
			{
				mMsServer = (AgentServerObjects.IAgentEx)new AgentServerObjects.AgentServer();
			}
			mMsServer.ShowDefaultCharacterProperties(0, 0, 1);
		}

		private void LoadMsControlChar_Click(object sender, EventArgs e)
		{
			if (mMsControlChar == null)
			{
				mMsControl.RaiseRequestErrors = false;
				mMsControl.Characters.Load(mMsCharacterName, mMsCharacterFile);
				mMsControl.RaiseRequestErrors = true;
				mMsControlChar = mMsControl.Characters.Character(mMsCharacterName);
				if (mMsControlChar != null)
				{
					mMsControlChar.MoveTo(700, 100, null);
				}
			}
			SetMsControlButtons();
		}

		private void UnloadMsControlChar_Click(object sender, EventArgs e)
		{
			if (mMsControlChar != null)
			{
				mMsControlChar = null;
				mMsControl.Characters.Unload(mMsCharacterName);
			}
			SetMsControlButtons();
		}

		private void ShowMsControlChar_Click(object sender, EventArgs e)
		{
			if (mMsControlChar != null)
			{
				mMsControlChar.Show(false);
				mMsControlChar.Speak(mMsControlChar.Name + ", Microsoft Agent", null);
			}
		}

		private void HideMsControlChar_Click(object sender, EventArgs e)
		{
			if (mMsControlChar != null)
			{
				mMsControlChar.Hide(false);
			}
		}

		private void ShowMsControlOptions_Click(object sender, EventArgs e)
		{
			mMsControl.PropertySheet.Visible = true;
		}

		private void ShowMsControlChars_Click(object sender, EventArgs e)
		{
			mMsControl.ShowDefaultCharacterProperties();
		}

		private void LoadDaServerChar_Click(object sender, EventArgs e)
		{
			int lReqId;

			if (mDaServer == null)
			{
				mDaServer = new DoubleAgent.Server.Server();
			}

			if (mDaServerChar == null)
			{
				mDaServer.Load(mDaCharacterFile, out mDaServerCharId, out lReqId);
                mDaServerChar = mDaServer.get_Character(mDaServerCharId);
				if (mDaServerChar != null)
				{
					mDaServerChar.MoveTo(700, 300, 0, out lReqId);
				}
			}
			SetDaServerButtons();
		}

		private void UnloadDaServerChar_Click(object sender, EventArgs e)
		{
			if ((mDaServer != null) && (mDaServerChar != null))
			{
				mDaServerChar = null;
				mDaServer.Unload(mDaServerCharId);
				mDaServerCharId = 0;
			}
			SetDaServerButtons();
		}

		private void ShowDaServerChar_Click(object sender, EventArgs e)
		{
			int lReqId;

			if (mDaServerChar != null)
			{
				mDaServerChar.Show(0, out lReqId);
				mDaServerChar.Speak(mDaCharacterName + ", Double Agent", "", out lReqId);
			}
			SetDaServerButtons();
		}

		private void HideDaServerChar_Click(object sender, EventArgs e)
		{
			int lReqId;

			if (mDaServerChar != null)
			{
				mDaServerChar.Hide(0, out lReqId);
			}
			SetDaServerButtons();
		}

		private void ShowDaServerOptions_Click(object sender, EventArgs e)
		{
			DoubleAgent.Server.PropertySheet lServerProps;

			if (mDaServer == null)
			{
				mDaServer = new DoubleAgent.Server.Server();
			}
			lServerProps = mDaServer.PropertySheet;
			lServerProps.SetVisible(1);
		}

		private void ShowDaServerChars_Click(object sender, EventArgs e)
		{
			if (mDaServer == null)
			{
				mDaServer = new DoubleAgent.Server.Server();
			}
			mDaServer.ShowDefaultCharacterProperties(0, 0, 1);
		}

		private void LoadDaControlChar_Click(object sender, EventArgs e)
		{
			if (mDaControlChar == null)
			{
				mDaControl.RaiseRequestErrors = false;
				mDaControl.Characters.Load(mDaCharacterName, mDaCharacterFile);
				mDaControl.RaiseRequestErrors = true;
				mDaControlChar = mDaControl.Characters.Character(mDaCharacterName);
				if (mDaControlChar != null)
				{
					mDaControlChar.MoveTo(700, 300, null);
				}
			}
			SetDaControlButtons();
		}

		private void UnloadDaControlChar_Click(object sender, EventArgs e)
		{
			if (mDaControlChar != null)
			{
				mDaControlChar = null;
				mDaControl.Characters.Unload(mDaCharacterName);
			}
			SetDaControlButtons();
		}

		private void ShowDaControlChar_Click(object sender, EventArgs e)
		{
			if (mDaControlChar != null)
			{
				mDaControlChar.Show(false);
				mDaControlChar.Speak(mDaControlChar.Name + ", Double Agent", null);
			}
		}

		private void HideDaControlChar_Click(object sender, EventArgs e)
		{
			if (mDaControlChar != null)
			{
				mDaControlChar.Hide(false);
			}
		}

		private void ShowDaControlOptions_Click(object sender, EventArgs e)
		{
			mDaControl.PropertySheet.Visible = true;
		}

		private void ShowDaControlChars_Click(object sender, EventArgs e)
		{
			mDaControl.ShowDefaultCharacterProperties(null, null);
		}

        private void aboutToolStripMenuItem_Click(object sender, EventArgs e)
        {
            AboutBox lAboutBox = new AboutBox();
            lAboutBox.mDaControl = mDaControl;
            lAboutBox.ShowDialog(this);
        }
	}
}

File: source/tags/alpha/build 1.3.0.57/Samples/VB/VS2010/MsaWithDa (Component)/MsaWithDa.vb
Option Strict Off
Option Explicit On
Friend Class MsaWithDa
    Inherits System.Windows.Forms.Form

    Private Const mMsCharacterName As String = "Genie"
    Private Const mMsCharacterFile As String = "C:\Windows\MsAgent\Chars\Genie.acs"

    Private Const mDaCharacterName As String = "Merlin"
    Private Const mDaCharacterFile As String = "C:\Windows\MsAgent\Chars\Merlin.acs"

    Dim mMsControlChar As AgentObjects.IAgentCtlCharacterEx
    Dim WithEvents mDaAgentControl As DoubleAgent.Control.Control
    Dim mDaControlChar As DoubleAgent.Control.Character

    Dim mMsServer As AgentServerObjects.AgentServer
    Dim mMsServerChar As AgentServerObjects.IAgentCharacterEx
    Dim mMsServerCharId As Integer

    Dim mDaServer As DoubleAgent.Server.Server
    Dim mDaServerChar As DoubleAgent.Server.Character
    Dim mDaServerCharId As Integer

    Private Sub MsaWithDa_Load(sender As System.Object, e As System.EventArgs) Handles MyBase.Load
        mDaAgentControl = New DoubleAgent.Control.Control
        SetDaControlButtons()
        SetMsControlButtons()
        SetMsServerButtons()
        SetDaControlButtons()
        SetDaServerButtons()
    End Sub

    Private Sub MsaWithDa_FormClosed(sender As System.Object, e As System.Windows.Forms.FormClosedEventArgs) Handles Me.FormClosed
        mDaAgentControl.Connected = False
        mMsAgentControl.Connected = False
    End Sub

    '***********************************************************************
    Private Sub SetMsControlButtons()
        If (mMsControlChar Is Nothing) Then
            LoadMsControlChar.Enabled = True
            UnloadMsControlChar.Enabled = False
            ShowMsControlChar.Enabled = False
            HideMsControlChar.Enabled = False
        Else
            LoadMsControlChar.Enabled = False
            UnloadMsControlChar.Enabled = True
            ShowMsControlChar.Enabled = Not mMsControlChar.Visible
            HideMsControlChar.Enabled = mMsControlChar.Visible
        End If
    End Sub

    Private Sub mMsAgentControl_HideEvent(sender As System.Object, e As AxAgentObjects._AgentEvents_HideEvent) Handles mMsAgentControl.HideEvent
        SetMsControlButtons()
    End Sub

    Private Sub mMsAgentControl_ShowEvent(sender As System.Object, e As AxAgentObjects._AgentEvents_ShowEvent) Handles mMsAgentControl.ShowEvent
        SetMsControlButtons()
    End Sub

    Private Sub LoadMsControlChar_Click(sender As System.Object, e As System.EventArgs) Handles LoadMsControlChar.Click
        If (mMsControlChar Is Nothing) Then
            mMsAgentControl.RaiseRequestErrors = False
            mMsAgentControl.Characters.Load(mMsCharacterName, mMsCharacterFile)
            mMsAgentControl.RaiseRequestErrors = True
            mMsControlChar = mMsAgentControl.Characters.Character(mMsCharacterName)
            If (Not mMsControlChar Is Nothing) Then
                mMsControlChar.MoveTo(700, 100)
            End If
        End If
        SetMsControlButtons()
    End Sub

    Private Sub UnloadMsControlChar_Click(sender As System.Object, e As System.EventArgs) Handles UnloadMsControlChar.Click
        If Not (mMsControlChar Is Nothing) Then
            mMsControlChar = Nothing
            mMsAgentControl.Characters.Unload(mMsCharacterName)
        End If
        SetMsControlButtons()
    End Sub

    Private Sub ShowMsControlChar_Click(sender As System.Object, e As System.EventArgs) Handles ShowMsControlChar.Click
        If (Not mMsControlChar Is Nothing) Then
            mMsControlChar.Show(False)
            mMsControlChar.Speak(mMsControlChar.Name + ", Microsoft Agent", "")
        End If
    End Sub

    Private Sub HideMsControlChar_Click(sender As System.Object, e As System.EventArgs) Handles HideMsControlChar.Click
        If (Not mMsControlChar Is Nothing) Then
            mMsControlChar.Hide(False)
        End If
    End Sub

    Private Sub ShowMsControlOptions_Click(sender As System.Object, e As System.EventArgs) Handles ShowMsControlOptions.Click
        mMsAgentControl.PropertySheet.Visible = True
    End Sub

    Private Sub ShowMsControlChars_Click(sender As System.Object, e As System.EventArgs) Handles ShowMsControlChars.Click
        mMsAgentControl.ShowDefaultCharacterProperties()
    End Sub
    '***********************************************************************
    Private Sub SetDaControlButtons()
        If (mDaControlChar Is Nothing) Then
            LoadDaControlChar.Enabled = True
            UnloadDaControlChar.Enabled = False
            ShowDaControlChar.Enabled = False
            HideDaControlChar.Enabled = False
        Else
            LoadDaControlChar.Enabled = False
            UnloadDaControlChar.Enabled = True
            ShowDaControlChar.Enabled = Not mDaControlChar.Visible
            HideDaControlChar.Enabled = mDaControlChar.Visible
        End If
    End Sub

    Private Sub mDaAgentControl_Hide(CharacterID As String, Cause As DoubleAgent.Control.VisibilityCauseType) Handles mDaAgentControl.Hide
        SetDaControlButtons()
    End Sub

    Private Sub mDaAgentControl_Show(CharacterID As String, Cause As DoubleAgent.Control.VisibilityCauseType) Handles mDaAgentControl.Show
        SetDaControlButtons()
    End Sub

    Private Sub LoadDaControlChar_Click(sender As System.Object, e As System.EventArgs) Handles LoadDaControlChar.Click
        If (mDaControlChar Is Nothing) Then
            mDaAgentControl.RaiseRequestErrors = False
            mDaAgentControl.Characters.Load(mDaCharacterName, mDaCharacterFile)
            mDaAgentControl.RaiseRequestErrors = True
            mDaControlChar = mDaAgentControl.Characters.Character(mDaCharacterName)
            If (Not mDaControlChar Is Nothing) Then
                mDaControlChar.MoveTo(700, 300)
            End If
        End If
        SetDaControlButtons()
    End Sub

    Private Sub UnloadDaControlChar_Click1(sender As Object, e As System.EventArgs) Handles UnloadDaControlChar.Click
        If Not (mDaControlChar Is Nothing) Then
            mDaControlChar = Nothing
            mDaAgentControl.Characters.Unload(mDaCharacterName)
        End If
        SetDaControlButtons()
    End Sub

    Private Sub ShowDaControlChar_Click(sender As System.Object, e As System.EventArgs) Handles ShowDaControlChar.Click
        If (Not mDaControlChar Is Nothing) Then
            mDaControlChar.Show(False)
            mDaControlChar.Speak(mDaControlChar.Name + ", Double Agent", "")
        End If
    End Sub

    Private Sub HideDaControlChar_Click(sender As System.Object, e As System.EventArgs) Handles HideDaControlChar.Click
        If (Not mDaControlChar Is Nothing) Then
            mDaControlChar.Hide(False)
        End If
    End Sub

    Private Sub ShowDaControlOptions_Click(sender As System.Object, e As System.EventArgs) Handles ShowDaControlOptions.Click
        mDaAgentControl.PropertySheet.Visible = True
    End Sub

    Private Sub ShowDaControlChars_Click(sender As System.Object, e As System.EventArgs) Handles ShowDaControlChars.Click
        mDaAgentControl.ShowDefaultCharacterProperties(0, 0)
    End Sub
    '***********************************************************************
    Private Sub SetMsServerButtons()
        If (mMsServerChar Is Nothing) Then
            LoadMsServerChar.Enabled = True
            UnloadMsServerChar.Enabled = False
            ShowMsServerChar.Enabled = False
            HideMsServerChar.Enabled = False
        Else
            LoadMsServerChar.Enabled = False
            UnloadMsServerChar.Enabled = True
            ShowMsServerChar.Enabled = True
            HideMsServerChar.Enabled = True
        End If
    End Sub

    Private Sub LoadMsServerChar_Click(sender As System.Object, e As System.EventArgs) Handles LoadMsServerChar.Click
        Dim lReqId As Integer

        If (mMsServer Is Nothing) Then
            mMsServer = New AgentServerObjects.AgentServer
        End If

        If (mMsServerChar Is Nothing) Then
            mMsServer.Load(mMsCharacterFile, mMsServerCharId, lReqId)
            mMsServer.GetCharacter(mMsServerCharId, mMsServerChar)
            If (Not mMsServerChar Is Nothing) Then
                mMsServerChar.MoveTo(700, 100, 0, lReqId)
            End If
        End If
        SetMsServerButtons()
    End Sub

    Private Sub UnloadMsServerChar_Click(sender As System.Object, e As System.EventArgs) Handles UnloadMsServerChar.Click
        If Not (mMsServer Is Nothing) And Not (mMsServerChar Is Nothing) Then
            mMsServerChar = Nothing
            mMsServer.Unload(mMsServerCharId)
            mMsServerCharId = 0
        End If
        SetMsServerButtons()
    End Sub

    Private Sub ShowMsServerChar_Click(sender As System.Object, e As System.EventArgs) Handles ShowMsServerChar.Click
        Dim lReqId As Integer

        If (Not mMsServerChar Is Nothing) Then
            mMsServerChar.Show(False, lReqId)
            mMsServerChar.Speak(mMsCharacterName + ", Microsoft Agent", "", lReqId)
        End If
        SetMsServerButtons()
    End Sub

    Private Sub HideMsServerChar_Click(sender As System.Object, e As System.EventArgs) Handles HideMsServerChar.Click
        Dim lReqId As Integer

        If (Not mMsServerChar Is Nothing) Then
            mMsServerChar.Hide(False, lReqId)
        End If
        SetMsServerButtons()
    End Sub

    Private Sub ShowMsServerChars_Click(sender As System.Object, e As System.EventArgs) Handles ShowMsServerChars.Click
        Dim lServerEx As AgentServerObjects.IAgentEx

        If (mMsServer Is Nothing) Then
            mMsServer = New AgentServerObjects.AgentServer
        End If

        lServerEx = mMsServer
        lServerEx.ShowDefaultCharacterProperties(0, 0, True)
    End Sub

    Private Sub ShowMsServerOptions_Click(sender As System.Object, e As System.EventArgs) Handles ShowMsServerOptions.Click
        Dim lServerProps As AgentServerObjects.IAgentPropertySheet

        If (mMsServer Is Nothing) Then
            mMsServer = New AgentServerObjects.AgentServer
        End If

        lServerProps = mMsServer
        lServerProps.SetVisible(True)
    End Sub
    '***********************************************************************
    Private Sub SetDaServerButtons()
        If (mDaServerChar Is Nothing) Then
            LoadDaServerChar.Enabled = True
            UnloadDaServerChar.Enabled = False
            ShowDaServerChar.Enabled = False
            HideDaServerChar.Enabled = False
        Else
            LoadDaServerChar.Enabled = False
            UnloadDaServerChar.Enabled = True
            ShowDaServerChar.Enabled = True
            HideDaServerChar.Enabled = True
        End If
    End Sub

    Private Sub LoadDaServerChar_Click(sender As System.Object, e As System.EventArgs) Handles LoadDaServerChar.Click
        Dim lReqId As Integer

        If (mDaServer Is Nothing) Then
            mDaServer = New DoubleAgent.Server.Server
        End If

        If (mDaServerChar Is Nothing) Then
            mDaServer.Load(mDaCharacterFile, mDaServerCharId, lReqId)
            mDaServerChar = mDaServer.Character(mDaServerCharId)
            If (Not mDaServerChar Is Nothing) Then
                mDaServerChar.MoveTo(700, 300, 0, lReqId)
            End If
        End If
        SetDaServerButtons()
    End Sub

    Private Sub UnloadDaServerChar_Click(sender As System.Object, e As System.EventArgs) Handles UnloadDaServerChar.Click
        If Not (mDaServer Is Nothing) And Not (mDaServerChar Is Nothing) Then
            mDaServerChar = Nothing
            mDaServer.Unload(mDaServerCharId)
            mDaServerCharId = 0
        End If
        SetDaServerButtons()
    End Sub

    Private Sub ShowDaServerChar_Click(sender As System.Object, e As System.EventArgs) Handles ShowDaServerChar.Click
        Dim lReqId As Integer

        If (Not mDaServerChar Is Nothing) Then
            mDaServerChar.Show(False, lReqId)
            mDaServerChar.Speak(mDaCharacterName + ", Double Agent", "", lReqId)
        End If
        SetDaServerButtons()
    End Sub

    Private Sub HideDaServerChar_Click(sender As System.Object, e As System.EventArgs) Handles HideDaServerChar.Click
        Dim lReqId As Integer

        If (Not mDaServerChar Is Nothing) Then
            mDaServerChar.Hide(False, lReqId)
        End If
        SetDaServerButtons()
    End Sub

    Private Sub ShowDaServerChars_Click(sender As System.Object, e As System.EventArgs) Handles ShowDaServerChars.Click
        Dim lServerProps As DoubleAgent.Server.PropertySheet

        If (mDaServer Is Nothing) Then
            mDaServer = New DoubleAgent.Server.Server
        End If

        lServerProps = mDaServer.PropertySheet
        lServerProps.Page = "Character"
        lServerProps.SetVisible(True)
    End Sub

    Private Sub ShowDaServerOptions_Click(sender As System.Object, e As System.EventArgs) Handles ShowDaServerOptions.Click
        Dim lServerProps As DoubleAgent.Server.PropertySheet

        If (mDaServer Is Nothing) Then
            mDaServer = New DoubleAgent.Server.Server
        End If

        lServerProps = mDaServer.PropertySheet
        lServerProps.SetVisible(True)
    End Sub

    Private Sub AboutToolStripMenuItem_Click(sender As System.Object, e As System.EventArgs) Handles AboutToolStripMenuItem.Click
        Dim lAbout As New AboutBox
        lAbout.mDaControl = mDaAgentControl
        lAbout.Show(Me)
    End Sub
End Class